tests/lax_numpy_test.py

```diff
@@ -1456,6 +1456,21 @@ def testDiag(self, shape, dtype, k, rng_factory):
     self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker, check_dtypes=True)
     self._CompileAndCheck(jnp_fun, args_maker, check_dtypes=True)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_shape={}_k={}".format(
+          jtu.format_shape_dtype_string(shape, dtype), k),
+       "dtype": dtype, "shape": shape, "k": k}
+      for dtype in default_dtypes
+      for shape in all_shapes
+      for k in range(-4, 4)))
+  def testDiagFlat(self, shape, dtype, k):
+    rng = jtu.rand_default(self.rng())
+    np_fun = lambda arg: np.diagflat(np.atleast_1d(arg), k)
```

Comment:
Add a comment specifying why we call ``atleast_1d`` for numpy but not jax; something like
```python
# numpy has inconsistencies for scalar values (see https://github.com/numpy/numpy/issues/16477)
# jax differs in that in all cases it treats scalar values of v as length-1 arrays.
```